cmd/edit/externalauthproviders/cmd.go

```diff
@@ -0,0 +1,162 @@
+	Use:     "external-auth-providers",
+	Aliases: []string{"externalauthproviders", "externalauthprovider", "external-auth-provider"},
+	Short:   "Edit external authentication providers",
+	Long:    "Edit external authentication providers on a cluster.",
+	Example: `  # Set issuer name 'issuer1' on cluster 'mycluster'
+  rosa edit external-auth-providers --issuer-name=issuer1 --cluster=mycluster`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	ocm.AddClusterFlag(Cmd)
+
+	flags.StringVar(
+		&args.name,
+		"name",
+		"",
+		"Name for the set of external authentication providers.",
+	)
+
+	flags.StringVar(
+		&args.issuerName,
+		"issuer-name",
+		"",
+		"Issuer name.",
+	)
+
+	flags.StringArrayVar(
+		&args.issuerAudiences,
+		"issuer-audiences",
+		nil,
+		"Names of audiences for the issuer.",
+	)
+
+	flags.StringVar(
+		&args.issuerUrl,
+		"issuer-url",
+		"",
+		"The url link for the issuer.",
+	)
+
+	// Is this a path?
+	flags.StringVar(
+		&args.issuerCaFile,
+		"issuer-ca-file",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimMappingGroupsClaim,
+		"claim-mapping-groups-claim",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimMappingUsernameClaim,
+		"claim-mapping-username-claim",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimValidationRuleClaim,
+		"claim-validation-rule-claim",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimValidationRuleRequiredValue,
+		"claim-validation-rule-required-value",
+		"",
+		"TBD.",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+
+	if cluster.Hypershift().Enabled() {
+
+		if cluster.State() != cmv1.ClusterStateReady {
+			r.Reporter.Errorf("Cluster '%s' is not yet ready. Current state is '%s'", clusterKey, cluster.State())
+			os.Exit(1)
+		}
+
+		externalAuthProviders, err := r.OCMClient.GetExternalAuthProviders(cluster.ID())
+		if err != nil {
+			r.Reporter.Errorf("Failed to fetch existing  external authentication providers for cluster '%s': %s",
+				clusterKey, err)
+			os.Exit(1)
+		}
+
+		if externalAuthProviders == nil {
+			r.Reporter.Errorf("No external authentication providers for cluster '%s' has been found. "+
+				"You should first create it via 'rosa create external-auth-providers'", clusterKey)
+			os.Exit(1)
+		}
+
+		r.Reporter.Debugf("Updating external authentication providers for cluster '%s'", clusterKey)
+
+		_, err = r.OCMClient.UpdateExternalAuthProviders(cluster.ID())
+		if err != nil {
+			r.Reporter.Errorf("Failed creating custom external authentication providers for cluster '%s': %s",
+				cluster.ID(), err)
+			os.Exit(1)
+		}
+
+		r.Reporter.Infof("Successfully updated custom external authentication providers for cluster '%s'", clusterKey)
+		os.Exit(0)
+
+	} else {
+		r.Reporter.Errorf("External authentication provider is only supported for Hosted Control Planes.")
+	}
```

Comment:
IMO would be best to move this to a new func that returns an error to make it easier to test :pray:

Reply:
🙏 To pass the coverage